extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -299,13 +363,158 @@ export class AmpStoryQuiz extends AMP.BaseElement {
    * @private
    */
   handleOptionSelection_(optionEl) {
-    this.triggerAnalytics_(optionEl);
+    this.responseDataPromise_.then(() => {
+      if (this.hasUserSelection_) {
+        return;
+      }
 
-    this.mutateElement(() => {
-      optionEl.classList.add('i-amphtml-story-quiz-option-selected');
-      this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+      this.triggerAnalytics_(optionEl);
+
+      this.mutateElement(() => {
+        optionEl.classList.add('i-amphtml-story-quiz-option-selected');
+        this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+
+        this.hasUserSelection_ = true;
+      });
+
+      if (this.element.hasAttribute('endpoint')) {
+        this.updateReactionData_(optionEl.optionIndex_);
+      }
+    });
+  }
+
+  /**
+   * Get the Reaction data from the datastore
+   *
+   * @return {Promise<JsonObject>}
+   * @private
+   */
+  retrieveReactionData_() {
+    return this.executeReactionRequest_(
+      dict({
+        'method': 'GET',
+      })
+    )
+      .then(response => {
+        this.handleSuccessfulDataRetrieval_(response);
+      })
+      .catch(error => {
+        dev().error(TAG, error);
+      });
+  }
+
+  /**
+   * Update the Reaction data in the datastore
+   *
+   * @param {number} reactionValue
+   * @private
+   */
+  updateReactionData_(reactionValue) {
+    this.executeReactionRequest_(
+      dict({
+        'method': 'POST',
+      }),
+      reactionValue
+    ).catch(error => {
+      dev().error(TAG, error);
+    });
+  }
 
-      this.hasReceivedResponse_ = true;
+  /**
+   * Executes a Reactions API call.
+   *
+   * @param {Object} requestOptions
+   * @param {number} reactionValue
+   * @return {Promise<JsonObject>}
+   * @private
+   */
+  executeReactionRequest_(requestOptions, reactionValue) {
+    // TODO(jackbsteinberg): Add a default reactions endpoint.
+    if (!assertAbsoluteHttpOrHttpsUrl(this.element.getAttribute('endpoint'))) {
+      return Promise.reject(ENDPOINT_INVALID_ERROR);
```

Comment:
Please also add a `dev` level log error

Reply:
I refactored the endpoint checks, meaning the requests are not triggered if there is no endpoint. 

Now this `ENDPOINT_INVALID_ERROR` is no longer ignored by the `.catch`, so it will be covered by the dev level log error in there.